gcc/rust/typecheck/rust-hir-type-check.cc

```diff
@@ -123,44 +123,25 @@ TypeResolution::Resolve (HIR::Crate &crate)
 void
 TypeCheckExpr::visit (HIR::BlockExpr &expr)
 {
-  TyTy::BaseType *block_tyty
-    = new TyTy::TupleType (expr.get_mappings ().get_hirid ());
-
   expr.iterate_stmts ([&] (HIR::Stmt *s) mutable -> bool {
-    bool is_final_stmt = expr.is_final_stmt (s);
-    bool has_final_expr = expr.has_expr () && expr.tail_expr_reachable ();
-    bool stmt_is_final_expr = is_final_stmt && !has_final_expr;
-
     auto resolved = TypeCheckStmt::Resolve (s, inside_loop);
     if (resolved == nullptr)
       {
 	rust_error_at (s->get_locus_slow (), "failure to resolve type");
 	return false;
       }
 
-    if (stmt_is_final_expr)
-      {
-	delete block_tyty;
-	block_tyty = resolved;
-      }
-    else if (!resolved->is_unit ())
-      {
```

Comment:
We still need this check for unit type for the case of:

```
fn test() -> i32 {
    if (...) { 123} else { 456 }
    789
}
```

Reply:
Now it is checked in https://github.com/Rust-GCC/gccrs/blob/3c251b99b80c05a7b655aa41b3285b7ce5215ede/gcc/rust/typecheck/rust-hir-type-check-stmt.h#L46-L50
since the following code is correct.
```rust
fn test() {
   123;
}
```